pkg/codegen/docs/gen.go

```diff
@@ -985,6 +1001,7 @@ func (mod *modContext) genResource(r *schema.Resource) resourceDocArgs {
 		if r.StateInputs != nil {
 			stateInputs[lang] = mod.getProperties(r.StateInputs.Properties, lang, true, false)
 		}
+		outputProps[lang] = filterOutputProperties(inputProps[lang], outputProps[lang])
```

Comment:
Instead of filtering the output properties after calling `getProperties` for the full list of output properties, could we do it before? That way, we just call `getProperties` with the filtered list of output properties? `getProperties` calls out to each lang code gen to generate a property's type string name for every language, including some string processing to strip namespaces etc. So in terms of time spent, it would be better to avoid doing all that for _all_ the output properties and just do that for only the properties we need to.

Reply:
ahh ok. I think I get what you are saying.

Reply:
Ok. I made the adjustment.... I think that is what you were thinking of.